tests/e2e/tidbcluster/serial.go

```diff
@@ -296,7 +296,7 @@ var _ = ginkgo.Describe("[Serial]", func() {
 			oa.UpgradeOperatorOrDie(ocfg)
 			// now the webhook enabled
 			err = controller.GuaranteedUpdate(genericCli, legacyTc, func() error {
-				legacyTc.Spec.TiDB.Image = fmt.Sprintf("pingcap/tidb:%s", utilimage.TiDBV3UpgradeVersion)
+				legacyTc.Spec.TiDB.Image = fmt.Sprintf("pingcap/tidb:%s", utilimage.TiDBV4)
```

Comment:
No image update in this step.

Reply:
fixed in [3ced58a](https://github.com/pingcap/tidb-operator/pull/3770/commits/3ced58a84bb36a0a45043f19337f83722a60cc5f)